website/docs/plugin/framework/ephemeral-resources/renew.mdx

```diff
@@ -0,0 +1,113 @@
+
+In this example, an ephemeral resource named `examplecloud_thing` with hardcoded behavior is defined. It indicates a renewal should occur 5 minutes from when either the `Open` or `Renew` method is executed:
+
+```go
+var _ ephemeral.EphemeralResourceWithRenew = (*ThingEphemeralResource)(nil)
+
+// ThingEphemeralResource defines the ephemeral resource implementation, which also implements Renew.
+type ThingEphemeralResource struct{}
+
+type ThingEphemeralResourceModel struct {
+	Name  types.String `tfsdk:"name"`
+	Token types.String `tfsdk:"token"`
+}
+
+type ThingPrivateData struct {
+	Name string `json:"name"`
+}
+
+func (e *ThingEphemeralResource) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Description: "Name of the thing to retrieve a token for.",
+				Required:    true,
+			},
+			"token": schema.StringAttribute{
+				Description: "Token for the thing.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (e *ThingEphemeralResource) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data ThingEphemeralResourceModel
+
+	// Read Terraform config data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Typically ephemeral resources will make external calls and reference returned data,
+	// however this example hardcodes the setting of result and private data for brevity.
+	data.Token = types.StringValue("token-123")
+
+	// Renew 5 minutes from now
+	resp.RenewAt = time.Now().Add(5 * time.Minute)
+
+	// When renewing, pass along this data (error handling omitted for brevity).
+	privateData, _ := json.Marshal(ThingPrivateData{Name: data.Name.ValueString()})
+	resp.Private.SetKey(ctx, "thing_data", privateData)
+
+	// Save data into ephemeral result data
+	resp.Diagnostics.Append(resp.Result.Set(ctx, &data)...)
+}
+
+func (e *ThingEphemeralResource) Renew(ctx context.Context, req ephemeral.RenewRequest, resp *ephemeral.RenewResponse) {
+	privateBytes, _ := req.Private.GetKey(ctx, "thing_data")
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Unmarshal private data (error handling omitted for brevity).
+	var privateData ThingPrivateData
+	json.Unmarshal(privateBytes, &privateData)
+
+	// Perform external call to renew "thing" data
+
+	// Renew again in 5 minutes
+	resp.RenewAt = time.Now().Add(5 * time.Minute)
+
+	// If needed, you can also set new `Private` data on the response.
+}
+```
+
+## Recommendations
+
+* For the `RenewAt` response field, provide small leeway before an ephemeral resource expires, usually no more than a few minutes, to account for clock skew.
```

Comment:
Nit: this sentence might be too verbose and the audience might not know what `clock skew` means.

```suggestion
* When setting the `RenewAt` response field, add extra time (usually no more than a few minutes) before an ephemeral resource expires to account for latency.
```

Reply:
Good call, also updated the package docs ⚡ 370da65c406b2095707c4f2be60f512b9e97d796